server/src/main/java/org/apache/iotdb/db/sync/datasource/DeletionGroup.java

```diff
@@ -0,0 +1,274 @@
+  private TreeMap<Long, Long> delIntervalMap;
+
+  public enum DeletedType {
+    NO_DELETED,
+    PARTIAL_DELETED,
+    FULL_DELETED
+  }
+
+  public static class TsInterval {
+    public long startTime;
+    public long endTime;
+
+    public TsInterval(long startTime, long endTime) {
+      this.startTime = startTime;
+      this.endTime = endTime;
+    }
+
+    public void set(long startTime, long endTime) {
+      this.startTime = startTime;
+      this.endTime = endTime;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) {
+        return true;
+      }
+      if (!(obj instanceof TsInterval)) {
+        return false;
+      }
+
+      TsInterval tsInterval = (TsInterval) obj;
+      return (tsInterval.startTime == this.startTime) && (tsInterval.endTime == this.endTime);
+    }
+  }
+
+  public static class IntervalCursor {
+    Iterator<Map.Entry<Long, Long>> iter = null;
+    boolean subsequentNoDelete = false;
+    public long startTime;
+    public long endTime;
+
+    public void reset() {
+      iter = null;
+      subsequentNoDelete = false;
+    }
+  }
+
+  public DeletionGroup() {
+    delIntervalMap = new TreeMap<>();
+  }
+
+  /**
+   * Insert delete time interval data for very deletion.
+   *
+   * @param startTime
+   * @param endTime
+   */
+  public void addDelInterval(long startTime, long endTime) {
+    if (startTime > endTime) {
+      throw new IllegalArgumentException("addDelInterval(), error: startTime > endTime.");
+    }
+
+    // == pay attention, intervalMap's Entries are not overlap.
+    Map.Entry<Long, Long> startEntry = delIntervalMap.floorEntry(startTime);
+    Map.Entry<Long, Long> endEntry = delIntervalMap.floorEntry(endTime);
+
+    if ((startEntry != null) && (startTime <= startEntry.getValue())) {
+      startTime = startEntry.getKey();
+    }
+    if ((endEntry != null) && (endTime < endEntry.getValue())) {
+      endTime = endEntry.getValue();
+    }
+
+    // == find existing overlap entries and remove them
+    Map<Long, Long> overlapEntries = delIntervalMap.subMap(startTime, true, endTime, true);
+    Iterator<Map.Entry<Long, Long>> iter = overlapEntries.entrySet().iterator();
+    while (iter.hasNext()) {
+      iter.next();
+      iter.remove();
+    }
+    //    Long[] keyList = overlapEntries.keySet().toArray(new Long[0]);
+    //    for (Long key : keyList) {
+    //      delIntervalMap.remove(key);
+    //    }
+
+    delIntervalMap.put(startTime, endTime); // add new deletion interval
+  }
+
+  /**
+   * If this object has no deletion data (i.e delIntervalMap is empty), return true
+   *
+   * @return
+   */
+  public boolean isEmpty() {
+    return delIntervalMap.isEmpty();
+  }
+
+  /**
+   * Check the deletion-state of the data-points of specific time range according to the info of
+   * .mods
+   *
+   * @param startTime - the start time of data set, inclusive
+   * @param endTime - the end time of data set, inclusive
+   * @return
```

Comment:
Please add simple annotation for return value or `DeletedType`

Reply:
thanks.  I have just added the annotation  to the definition of DeletedType